src/history/hs.c

```diff
@@ -776,68 +778,56 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
-                prev_full_value->size = prev_upd->size;
-            } else
-                WT_ERR(__hs_calculate_full_value(
-                  session, prev_full_value, prev_upd, full_value->data, full_value->size));
+            WT_ERR(__hs_next_upd_full_value(
+              session, &modifies, full_value, btree->id, key, prev_full_value, &prev_upd));
 
             /*
-             * Skip the updates have the same start timestamp and transaction id
+             * Skip the updates have the same start timestamp and transaction id.
              *
              * Modifies that have the same start time point as the onpage_upd can be squashed away.
              */
-            if (upd->start_ts != prev_upd->start_ts || upd->txnid != prev_upd->txnid) {
-                /*
-                 * Calculate reverse delta. Insert full update for the newest historical record even
-                 * it's a MODIFY.
-                 *
-                 * It is not correct to check prev_upd == list->onpage_upd as we may have aborted
-                 * updates in the middle.
-                 *
-                 * We must insert the first and second updates after a prepared update as full
-                 * values because if the prepared update is aborted, we will remove the first update
-                 * after it from the history store to the update chain. Readers reading the older
-                 * values need a full update as the base value for constructing reverse modifies.
-                 */
-                nentries = MAX_REVERSE_MODIFY_NUM;
-                if (!F_ISSET(upd, WT_UPDATE_HS)) {
-                    if (upd->type == WT_UPDATE_MODIFY &&
-                      prev_upd->prepare_state != WT_PREPARE_INPROGRESS &&
-                      (second_older_than_prepare == NULL || upd != second_older_than_prepare) &&
-                      __wt_calc_modify(session, prev_full_value, full_value,
-                        prev_full_value->size / 10, entries, &nentries) == 0) {
-                        WT_ERR(__wt_modify_pack(cursor, entries, nentries, &modify_value));
-                        WT_ERR(__hs_insert_record(session, cursor, btree, key, upd,
-                          WT_UPDATE_MODIFY, modify_value, &stop_time_point));
-                        __wt_scr_free(session, &modify_value);
-                    } else
-                        WT_ERR(__hs_insert_record(session, cursor, btree, key, upd,
-                          WT_UPDATE_STANDARD, full_value, &stop_time_point));
-
-                    /* Flag the update as now in the history store. */
-                    F_SET(upd, WT_UPDATE_HS);
-                    ++insert_cnt;
-                    if (squashed) {
-                        WT_STAT_CONN_INCR(session, cache_hs_write_squash);
-                        squashed = false;
-                    }
-                }
-            } else
+            if (upd->start_ts == prev_upd->start_ts && upd->txnid == prev_upd->txnid) {
```

Comment:
In future if we want to insert reverse modifies for squashed transactions, we can implement the logic in `__hs_next_upd_full_value `